Project/Controllers/ToDoController.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Project.Models;
+
+namespace Project.Controllers
+{
+    [Route("todo")]
+    [ApiController]
+    public class ToDoController : ControllerBase
+    {
+        private readonly ToDoContext _context;
+
+        public ToDoController(ToDoContext context) => _context = context;
+
+        private StatusMethods _code = new();
+
+        [HttpPost]
+        [ProducesDefaultResponseType]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        public ActionResult<ToDo> PostItem(ToDo todo)
+        {
+            try
+            {
+                if (String.IsNullOrEmpty(todo.Description))
+                {
+                    return BadRequest("Description field is null or empty.");
+                }
+
+                _context.ToDoItems.Add(todo);
+                _context.SaveChanges();
+
+                return CreatedAtAction("GetItem", new ToDo {Id = todo.Id}, todo);
+            }
+            catch (Exception e)
```

Comment:
Ok let's take this one by one.

What can go wrong here?
- validation issue
- some sort of database connection issue

For now let's just handle the validation issue, we want to return `400` Bad Request and tell the user that validation failed.